pkg/generic/pbidl_provider.go

```diff
@@ -97,3 +107,29 @@ func (p *PbContentProvider) Close() error {
 	})
 	return nil
 }
+
+// PbContentProviderDynamicGo
+func NewPbContentProviderDynamicGo(main string, ctx context.Context, options dproto.Options) (PbDescriptorProviderDynamicGo, error) {
```

Comment:
Nits: There are two interfaces to get a pb provider, `NewDescriptorFromPath` and `NewDescriptorFromContent`.
You use `NewDescriptorFromPath`, so I think `NewPbFileProviderWithDynamicGo` would be better to understand. (you can refer to thriftidl_provider.go)